### PR TITLE
feature: replace custom Markdown editor with EasyMDE

### PR DESCRIPTION
**Pull request by ryangjchandler**
This pull request aims to replace our custom Markdown editor with something a little more battle tested, EasyMDE.

The process here is quite simple, EasyMDE is more flexible than I had originally thought. At the time of writing this, the editor will do almost exactly the same thing as before (text formatting, insert image and link wrappers, preview markdown). Below is a list of things that still need to be done.

**TODO:**
* [x] Get EasyMDE setup
* [x] Make toolbar buttons do the same thing 
* [x] Get drag & drop image uploads working (take code from old component)
* [x] Get file picker image uploads working (use same code as above)
* [x] Fix preview pane not going full content height when toggled
* [x] Add heading button?
* [x] Dark mode
* [x] RTL

Quite a bit to do still, but it's functional right now.

### CODE REVIEW

File: packages/forms/resources/css/components/markdown-editor.css [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+.EasyMDEContainer .editor-toolbar {
+    @apply hidden;
+}
+
+.EasyMDEContainer .CodeMirror {
+    @apply w-full h-full min-h-full transition duration-75 rounded-lg shadow-sm border border-gray-300 focus:border-primary-500 focus:ring-1 focus:ring-inset focus:ring-primary-600 caret-black !important;
+}
+
+.EasyMDEContainer .CodeMirror .CodeMirror-scroll {
+    @apply tracking-normal whitespace-pre-wrap font-mono text-sm !important;
```
Comment (danharrin) [inline packages/forms/resources/css/components/markdown-editor.css:12]: !important does not work with @apply in Sass, so these need to be normal CSS declarations :(

**Comment by ralphjsmit** [comment]
Will this keep the same appearance as the custom Markdown editor?

**Comment by ryangjchandler** [comment]
@ralphjsmit Yeah, all looks the same! There's a new header button in the toolbar, but otherwise it's the same 😊

File: packages/forms/resources/css/components/markdown-editor.css (hunk 1 above)
Comment (ryangjchandler) [reply packages/forms/resources/css/components/markdown-editor.css:12]: Ah, that sucks. Important probably isn't even needed because of specificity.

**Comment by themahdavi** [comment]
Nice 👌
I tested and two bug :
1- dark mode not work for textarea
2- for RTL need direction property on `new EasyMDE()`

**Comment by ryangjchandler** [comment]
Thanks for testing! I'll sort those out today :)

**Comment by ryangjchandler** [comment]
@themahdavi Are you able to test again - I've verified dark mode fixes but would like to check RTL support.

**Comment by ziming** [comment]
The current markdown field look kind of ugly when ->disabled() is called on it. I hope this new one look nicer!

**Comment by ziming** [comment]
I see. Can the Preview mode not have that weird circle? @ryangjchandler 

https://share.getcloudapp.com/jkuX1Rnk

Thank you!

**Comment by ryangjchandler** [comment]
@ziming Yeah, I've fixed that in this pull request. I think that was a conflict between some padding classes.

**Comment by themahdavi** [comment]
@ryangjchandler Dark mode and RTL work correctly but two thing need fix:
1- on darkmode preview show preview box color smaller 
image

2- on RTL preview all html items go left and text on right
image

**Comment by themahdavi** [comment]
And it's Nice if showing (uploading image) text when image uploading progress...

**Comment by ryangjchandler** [comment]
Hm, the RTL problem on preview is down to EasyMDE..

**Comment by ryangjchandler** [comment]
@danharrin I think this is functionally ready. I might have to open an issue on the EasyMDE repo regarding the RTL problem with lists though.

**Comment by themahdavi** [comment]
@ryangjchandler how for implement uploading state like github when uploading in progress?
is it possible with easymde ?

**Comment by ryangjchandler** [comment]
Sorry missed your comment about that. Good idea. We can definitely do something similar to the file upload fields where we set the buttons to disabled with "file is uploading" text. We can also add a small status bar to the Markdown editor perhaps?

**Comment by themahdavi** [comment]
@ryangjchandler yes status bar is good but i think have state on button or text like github is better

**Comment by themahdavi** [comment]
And i think increase dependencies of project is maybe reduce control ...
smallest MarkdownEditor is perfect with simple features
and have seprate plugin is better
Of course this is my opinion

**Comment by ryangjchandler** [comment]
@themahdavi Using an existing solution takes some of the maintenance pressure off of us, especially when it does things that our existing editor couldn't do / didn't do.

**Comment by jeffgreco13** [comment]
@ryangjchandler how's this coming along? Just had a need to include headings. No pressure, but if this is imminent then I'll wait instead of using RichText :)

**Comment by danharrin** [comment]
Sorry @jeffgreco13, theres no block AFAIK, I am just slow at reviewing

**Comment by jeffgreco13** [comment]
@danharrin Hey no worries at all! Lots on your plate.

**Comment by ryangjchandler** [comment]
Headings will work if you use the regular MD syntax for them!

**Comment by ryangjchandler** [comment]
Needed a rebase - ready for review whenever :)

**Comment by danharrin** [comment]
Just reviewed this. It's perfect, except when you change the state from the server.

For example, when using a simple resource / relation manager, and switching between records.

To make the editor contents update, you have to click on it, scroll, and then it glitches in.

**Pull request closed without merging**